src/QMCDrivers/QMCDriver.h

```diff
@@ -187,7 +187,7 @@ class QMCDriver : public QMCDriverInterface, public QMCTraits, public MPIObjectB
   std::unique_ptr<TraceManager> Traces;
 
   ///return the random generators
-  inline std::vector<RandomGenerator_t*>& getRng() { return Rng; }
+  inline UPtrVector<RandomGenerator_t>& getRng() { return Rng; }
```

Comment:
Do we want to change this return type? Maintaining an owning and non-owning vector adds complexity but improves encapsulation.

If we're changing everything back anyway, a vector of references might be superior to a vector of pointers.

Reply:
@quantumsteve  `std::vector<RandomGenerator_t&>` is not possible, I think there is `std::reference_wrapper` but pros and cons should be discussed (I don't know the overall context and I never used `std::reference_wrapper`, but either smart of raw pointers).

Reply:
@williamfgc Yes, std::reference_wrapper is used elsewhere in qmcpack.
https://github.com/QMCPACK/qmcpack/blob/4c5968fbf58a83794cb8b714ae691b03ccbb74dc/src/type_traits/template_types.hpp#L60

Lots of ways to get the same behavior.